diff -up ./Makefile.fixsanity ./Makefile
--- ./Makefile.fixsanity	2024-07-26 00:35:57.849070455 +0530
+++ ./Makefile	2024-07-26 00:36:27.996901984 +0530
@@ -53,7 +53,6 @@ bindata: ## Update project bindata

 .PHONY: fix
 fix: ## Fixup files in the repo.
-	$(GO) mod tidy
 	$(GO) fmt ./...
 	make setup-lint
 	$(TOOLS_DIR)/golangci-lint run --fix --build-tags $(GO_BUILD_TAGS)
@@ -149,14 +148,10 @@ test-all: test-static test-e2e ## Run all tests
 test-static: test-sanity test-unit test-docs ## Run all non-cluster-based tests

 .PHONY: test-sanity
-test-sanity: generate fix ## Test repo formatting, linting, etc.
-	git diff --exit-code # fast-fail if generate or fix produced changes
+test-sanity: build fix lint ## Test repo formatting, linting, etc.
 	./hack/check-license.sh
 	./hack/check-error-log-msg-format.sh
 	$(GO) vet -tags=$(GO_BUILD_TAGS) ./...
-	make setup-lint
-	make lint
-	git diff --exit-code # diff again to ensure other checks don't change repo

 .PHONY: test-docs
 test-docs: ## Test doc links

diff -up ./hack/lib/common.sh.fixsanity ./hack/lib/common.sh
--- ./hack/lib/common.sh.fixsanity      2025-07-04 18:54:02.567064328 +0530
+++ ./hack/lib/common.sh        2025-07-04 18:53:57.896096562 +0530
@@ -57,7 +57,7 @@ function trap_add() {
 }

 function listPkgDirs() {
-	go list -f '{{.Dir}}' ./cmd/... ./test/... ./internal/... | grep -v generated
+	go list -tags=containers_image_openpgp -f '{{.Dir}}' ./cmd/... ./test/... ./internal/... | grep -v generated
 }

 function listFiles() {
