diff -up ./Makefile.fixSanity ./Makefile
--- ./Makefile.fixSanity	2021-04-28 09:32:31.944383270 -0700
+++ ./Makefile	2021-04-28 09:35:28.386754959 -0700
@@ -48,7 +48,6 @@ bindata: ## Update project bindata
 
 .PHONY: fix
 fix: ## Fixup files in the repo.
-	go mod tidy
 	go fmt ./...
 
 .PHONY: clean
@@ -118,13 +117,11 @@ test-all: test-static test-e2e ## Run al
 test-static: test-sanity test-unit test-docs ## Run all non-cluster-based tests
 
 .PHONY: test-sanity
-test-sanity: generate fix ## Test repo formatting, linting, etc.
-	git diff --exit-code # fast-fail if generate or fix produced changes
+test-sanity: build fix ## Test repo formatting, linting, etc.
 	./hack/check-license.sh
 	./hack/check-error-log-msg-format.sh
 	go vet ./...
 	$(SCRIPTS_DIR)/fetch golangci-lint 1.31.0 && $(TOOLS_DIR)/golangci-lint run
-	git diff --exit-code # diff again to ensure other checks don't change repo
 
 .PHONY: test-docs
 test-docs: ## Test doc links
