diff -up ./Makefile.fixsanity ./Makefile
--- ./Makefile.fixsanity	2024-07-26 00:35:57.849070455 +0530
+++ ./Makefile	2024-07-26 00:36:27.996901984 +0530
@@ -51,7 +51,6 @@ bindata: ## Update project bindata
 
 .PHONY: fix
 fix: ## Fixup files in the repo.
-	$(GO) mod tidy
 	$(GO) fmt ./...
 	make setup-lint
 	$(TOOLS_DIR)/golangci-lint run --fix
@@ -140,14 +139,10 @@ test-all: test-static test-e2e ## Run al
 test-static: test-sanity test-unit test-docs ## Run all non-cluster-based tests
 
 .PHONY: test-sanity
-test-sanity: generate fix ## Test repo formatting, linting, etc.
-	git diff --exit-code # fast-fail if generate or fix produced changes
+test-sanity: build fix lint ## Test repo formatting, linting, etc.
 	./hack/check-license.sh
 	./hack/check-error-log-msg-format.sh
 	$(GO) vet ./...
-	make setup-lint
-	make lint
-	git diff --exit-code # diff again to ensure other checks don't change repo
 
 .PHONY: test-docs
 test-docs: ## Test doc links
