diff -up ./Makefile.fixsanity ./Makefile
--- ./Makefile.fixsanity	2020-12-16 17:14:41.501091098 -0500
+++ ./Makefile	2020-12-16 17:18:02.147806579 -0500
@@ -47,7 +47,6 @@ bindata: ## Update project bindata
 
 .PHONY: fix
 fix: ## Fixup files in the repo.
-	go mod tidy
 	go fmt ./...
 
 .PHONY: clean
@@ -99,14 +98,12 @@ test-all: test-static test-e2e ## Run al
 test-static: test-sanity test-unit test-links ## Run all non-cluster-based tests
 
 .PHONY: test-sanity
-test-sanity: generate fix ## Test repo formatting, linting, etc.
-	git diff --exit-code # fast-fail if generate or fix produced changes
+test-sanity: build fix ## Test repo formatting, linting, etc.
 	./hack/check-license.sh
 	./hack/check-error-log-msg-format.sh
 	go run ./release/changelog/gen-changelog.go -validate-only
 	go vet ./...
 	$(SCRIPTS_DIR)/fetch golangci-lint 1.31.0 && $(TOOLS_DIR)/golangci-lint run
-	git diff --exit-code # diff again to ensure other checks don't change repo
 
 .PHONY: test-links
 test-links: ## Test doc links
