diff -up ./Makefile.fixsanity ./Makefile
--- ./Makefile.fixsanity	2022-06-15 23:45:39.660734180 -0400
+++ ./Makefile	2022-06-15 23:47:58.464081489 -0400
@@ -49,7 +49,6 @@ bindata: ## Update project bindata
 
 .PHONY: fix
 fix: ## Fixup files in the repo.
-	go mod tidy
 	go fmt ./...
 
 .PHONY: clean
@@ -127,13 +126,11 @@ test-all: test-static test-e2e ## Run al
 test-static: test-sanity test-unit test-docs ## Run all non-cluster-based tests
 
 .PHONY: test-sanity
-test-sanity: generate fix ## Test repo formatting, linting, etc.
-	git diff --exit-code # fast-fail if generate or fix produced changes
+test-sanity: build fix ## Test repo formatting, linting, etc.
 	./hack/check-license.sh
 	./hack/check-error-log-msg-format.sh
 	go vet ./...
 	$(SCRIPTS_DIR)/fetch golangci-lint 1.46.2 && $(TOOLS_DIR)/golangci-lint run
-	git diff --exit-code # diff again to ensure other checks don't change repo
 
 .PHONY: test-docs
 test-docs: ## Test doc links
