diff -up ./Makefile.setversion ./Makefile
--- ./Makefile.setversion	2024-08-22 12:10:33.470623068 +0530
+++ ./Makefile	2024-08-22 12:11:12.739726012 +0530
@@ -6,8 +6,8 @@ SHELL = /bin/bash
 # version is moved to a separate repo and release process.
 export IMAGE_VERSION = v1.36.1
 # Build-time variables to inject into binaries
-export SIMPLE_VERSION = $(shell (test "$(shell git describe --tags)" = "$(shell git describe --tags --abbrev=0)" && echo $(shell git describe --tags)) || echo $(shell git describe --tags --abbrev=0)+git)
-export GIT_VERSION = $(shell git describe --dirty --tags --always)
+export SIMPLE_VERSION = v1.39.1-ocp
+export GIT_VERSION = $(SIMPLE_VERSION)
 export GIT_COMMIT = $(shell git rev-parse HEAD)
 export K8S_VERSION = 1.29.0
 
