diff -up ./Makefile.setversion ./Makefile
--- ./Makefile.setversion	2021-01-18 18:00:48.798480030 -0500
+++ ./Makefile	2021-01-18 18:01:39.492178121 -0500
@@ -6,8 +6,8 @@ SHELL = /bin/bash
 # version is moved to a separate repo and release process.
 export IMAGE_VERSION = v1.3.0
 # Build-time variables to inject into binaries
-export SIMPLE_VERSION = $(shell (test "$(shell git describe)" = "$(shell git describe --abbrev=0)" && echo $(shell git describe)) || echo $(shell git describe --abbrev=0)+git)
-export GIT_VERSION = $(shell git describe --dirty --tags --always)
+export SIMPLE_VERSION = v1.20.0-ocp
+export GIT_VERSION = $(SIMPLE_VERSION)
 export GIT_COMMIT = $(shell git rev-parse HEAD)
 export K8S_VERSION = 1.19.4
 
