diff -up ./Makefile.setversion ./Makefile
--- ./Makefile.setversion	2022-06-15 23:51:25.351598627 -0400
+++ ./Makefile	2022-06-15 23:53:16.378875096 -0400
@@ -6,8 +6,8 @@ SHELL = /bin/bash
 # version is moved to a separate repo and release process.
 export IMAGE_VERSION = v1.22.0
 # Build-time variables to inject into binaries
-export SIMPLE_VERSION = $(shell (test "$(shell git describe --tags)" = "$(shell git describe --tags --abbrev=0)" && echo $(shell git describe --tags)) || echo $(shell git describe --tags --abbrev=0)+git)
-export GIT_VERSION = $(shell git describe --dirty --tags --always)
+export SIMPLE_VERSION = v1.25.0-ocp
+export GIT_VERSION = $(SIMPLE_VERSION)
 export GIT_COMMIT = $(shell git rev-parse HEAD)
 export K8S_VERSION = 1.24.1
 
