diff -up ./internal/olm/operator/registry/operator_installer.go.bug1921458 ./internal/olm/operator/registry/operator_installer.go
--- ./internal/olm/operator/registry/operator_installer.go.bug1921458	2021-01-29 10:04:18.358090742 -0500
+++ ./internal/olm/operator/registry/operator_installer.go	2021-01-29 10:05:27.578027432 -0500
@@ -16,6 +16,7 @@ package registry
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"time"
 
@@ -108,6 +109,11 @@ func (o OperatorInstaller) UpgradeOperat
 		return nil, fmt.Errorf("error getting list of subscriptions: %v", err)
 	}
 
+	// If there are no subscriptions found, then the previous operator version doesn't exist, so return error
+	if len(subList.Items) == 0 {
+		return nil, errors.New("no existing operator found in the cluster to upgrade")
+	}
+
 	var subscription *v1alpha1.Subscription
 	for i := range subList.Items {
 		s := subList.Items[i]
