diff -up ./internal/olm/operator/registry/operator_installer.go.handle-sub-pkg-name-mismatch ./internal/olm/operator/registry/operator_installer.go
--- ./internal/olm/operator/registry/operator_installer.go.handle-sub-pkg-name-mismatch	2021-01-29 11:06:12.280775956 -0500
+++ ./internal/olm/operator/registry/operator_installer.go	2021-01-29 11:06:22.426912376 -0500
@@ -123,6 +123,10 @@ func (o OperatorInstaller) UpgradeOperat
 		}
 	}
 
+	if subscription == nil {
+		return nil, fmt.Errorf("subscription for package %q not found", o.PackageName)
+	}
+
 	log.Infof("Found existing subscription with name %s and namespace %s", subscription.Name, subscription.Namespace)
 
 	// todo: attempt #1 to trigger install plan for the subscription and
