diff -up ./Makefile.darwin ./Makefile
--- ./Makefile.darwin	2024-07-25 17:30:44.164036603 +0530
+++ ./Makefile	2024-07-25 17:32:13.983235216 +0530
@@ -208,3 +208,9 @@ help: ## Show this help screen.
 	@echo 'Available targets are:'
 	@echo ''
 	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-25s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+.PHONY: build-darwin
+build-darwin: ## Build operator-sdk.
+	@mkdir -p $(BUILD_DIR)/darwin_amd64 $(BUILD_DIR)/darwin_arm64
+	GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_ARGS) -o $(BUILD_DIR)/darwin_amd64/ ./cmd/operator-sdk
+	GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_ARGS) -o $(BUILD_DIR)/darwin_arm64/ ./cmd/operator-sdk
