unchanged:
--- internal/annotations/metrics/metrics.go.fix-stamps	2021-03-26 13:16:11.000000000 -0700
+++ internal/annotations/metrics/metrics.go	2021-03-26 13:19:43.000000000 -0700
@@ -69,6 +69,11 @@
 		return "unknown"
 	}
 
+	if strings.Contains(input, "ocp") {
+		version = version + "-ocp"
+		input = strings.Replace(input, "-ocp", "", 1)
+	}
+
 	if isUnreleased(input) {
 		version = version + "+git"
 	}
only in patch2:
unchanged:
--- internal/annotations/metrics/metrics_test.go.fix-stamps	2021-03-26 13:15:49.000000000 -0700
+++ internal/annotations/metrics/metrics_test.go	2021-03-26 13:16:52.000000000 -0700
@@ -15,6 +15,8 @@
 package metrics
 
 import (
+	"testing"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
@@ -36,6 +38,26 @@
 			output := parseVersion(version)
 			Expect(output).To(Equal("v0.18.0+git"))
 		})
+		It("should extract the right downstream sdk version", func() {
+			version := "v1.3.0-ocp"
+			output := parseVersion(version)
+			Expect(output).To(Equal("v1.3.0-ocp"))
+		})
+		It("should extract the right downstream sdk version", func() {
+			version := "v1.3.0-ocp-ge87627f4"
+			output := parseVersion(version)
+			Expect(output).To(Equal("v1.3.0-ocp+git"))
+		})
+		It("should extract the right downstream sdk version", func() {
+			version := "v1.3.0-ocp-ge87627f4-dirty"
+			output := parseVersion(version)
+			Expect(output).To(Equal("v1.3.0-ocp+git"))
+		})
 
 	})
 })
+
+func TestVersion(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "metrics Suite")
+}
