diff -up ./internal/cmd/operator-sdk/cli/cli.go.patchocpv1 ./internal/cmd/operator-sdk/cli/cli.go
--- ./internal/cmd/operator-sdk/cli/cli.go.patchocpv1	2024-12-26 18:02:08.422724303 +0530
+++ ./internal/cmd/operator-sdk/cli/cli.go	2024-12-26 18:04:50.221965067 +0530
@@ -42,6 +42,7 @@ import (
 	"github.com/operator-framework/operator-sdk/internal/plugins"
 	helmv1 "github.com/operator-framework/operator-sdk/internal/plugins/helm/v1"
 	manifestsv2 "github.com/operator-framework/operator-sdk/internal/plugins/manifests/v2"
+	openshiftv1 "github.com/operator-framework/operator-sdk/internal/plugins/openshift/v1"
 	scorecardv2 "github.com/operator-framework/operator-sdk/internal/plugins/scorecard/v2"
 	"github.com/operator-framework/operator-sdk/internal/util/projutil"
 )
@@ -78,6 +79,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *
 			golangv4.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
@@ -89,6 +91,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *
 			ansiblev1.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
@@ -100,6 +103,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *
 			helmv1.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
