diff -up ./internal/cmd/operator-sdk/cli/cli.go.patchocpv1 ./internal/cmd/operator-sdk/cli/cli.go
--- ./internal/cmd/operator-sdk/cli/cli.go.patchocpv1	2024-01-02 12:42:21
+++ ./internal/cmd/operator-sdk/cli/cli.go	2024-01-02 12:46:01
@@ -52,7 +52,9 @@ import (
 	envtestv1 "github.com/operator-framework/operator-sdk/internal/plugins/envtest/v1"
 	helmv1 "github.com/operator-framework/operator-sdk/internal/plugins/helm/v1"
 	manifestsv2 "github.com/operator-framework/operator-sdk/internal/plugins/manifests/v2"
+	openshiftv1 "github.com/operator-framework/operator-sdk/internal/plugins/openshift/v1"
 	scorecardv2 "github.com/operator-framework/operator-sdk/internal/plugins/scorecard/v2"
+
 	"github.com/operator-framework/operator-sdk/internal/util/projutil"
 )
 
@@ -102,6 +104,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command)
 			golangv3.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 		plugin.WithDeprecationMessage(golangv3.Plugin{}.DeprecationWarning()),
 	)
@@ -114,6 +117,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command)
 			golangv4.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
@@ -125,6 +129,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command)
 			ansiblev1.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
@@ -136,6 +141,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command)
 			helmv1.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
@@ -147,6 +153,7 @@ func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command)
 			hybrid.Plugin{},
 			manifestsv2.Plugin{},
 			scorecardv2.Plugin{},
+			openshiftv1.Plugin{},
 		),
 	)
 
