diff -up ./internal/cmd/operator-sdk/scorecard/cmd.go.bug2017630 ./internal/cmd/operator-sdk/scorecard/cmd.go
--- ./internal/cmd/operator-sdk/scorecard/cmd.go.bug2017630	2021-12-21 13:10:24.372692101 -0500
+++ ./internal/cmd/operator-sdk/scorecard/cmd.go	2021-12-21 13:11:49.311871742 -0500
@@ -141,6 +141,7 @@ func (c *scorecardCmd) convertXunit(outp
 			var tSuite xunit.TestSuite
 			tSuite.Name = res.Name
 			tCase.Name = res.Name
+			tCase.Classname = "scorecard"
 			if res.State == v1alpha3.ErrorState {
 				tCase.Errors = append(tCase.Errors, xunit.XUnitComplexError{Type: "Error", Message: strings.Join(res.Errors, ",")})
 				tSuite.Errors = strings.Join(res.Errors, ",")
