diff -up ./test/ansible/advanced-molecule-operator/Dockerfile.orig ./test/ansible/advanced-molecule-operator/Dockerfile
--- ./test/ansible/advanced-molecule-operator/Dockerfile.orig	2023-08-15 17:28:47.000000000 -0400
+++ ./test/ansible/advanced-molecule-operator/Dockerfile	2023-08-15 17:29:10.000000000 -0400
@@ -1,6 +1,6 @@
 FROM quay.io/operator-framework/ansible-operator:dev

 COPY requirements.yml ${HOME}/requirements.yml
-RUN ansible-galaxy collection install -r ${HOME}/requirements.yml \
+RUN ansible-galaxy collection install --force -r ${HOME}/requirements.yml \
  && chmod -R ug+rwx ${HOME}/.ansible

 COPY watches.yaml ${HOME}/watches.yaml