diff -up ./internal/plugins/ansible/v1/scaffolds/internal/templates/makefile.go.bug2037208 ./internal/plugins/ansible/v1/scaffolds/internal/templates/makefile.go
--- ./internal/plugins/ansible/v1/scaffolds/internal/templates/makefile.go.bug2037208	2022-01-07 14:38:48.458233200 -0500
+++ ./internal/plugins/ansible/v1/scaffolds/internal/templates/makefile.go	2022-01-07 14:41:05.104079002 -0500
@@ -113,7 +113,7 @@ undeploy: ## Undeploy controller from th
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
 OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
-ARCH := $(shell uname -m | sed 's/x86_64/amd64/')
+ARCH := $(shell uname -m | sed 's/x86_64/amd64/' | sed 's/aarch64/arm64/')
 
 .PHONY: kustomize
 KUSTOMIZE = $(shell pwd)/bin/kustomize
diff -up ./internal/plugins/helm/v1/scaffolds/internal/templates/makefile.go.bug2037208 ./internal/plugins/helm/v1/scaffolds/internal/templates/makefile.go
--- ./internal/plugins/helm/v1/scaffolds/internal/templates/makefile.go.bug2037208	2022-01-07 14:39:02.484422638 -0500
+++ ./internal/plugins/helm/v1/scaffolds/internal/templates/makefile.go	2022-01-07 14:41:40.998564124 -0500
@@ -113,7 +113,7 @@ undeploy: ## Undeploy controller from th
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
 OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
-ARCH := $(shell uname -m | sed 's/x86_64/amd64/')
+ARCH := $(shell uname -m | sed 's/x86_64/amd64/' | sed 's/aarch64/arm64/')
 
 .PHONY: kustomize
 KUSTOMIZE = $(shell pwd)/bin/kustomize
diff -up ./testdata/ansible/memcached-operator/Makefile.bug2037208 ./testdata/ansible/memcached-operator/Makefile
--- ./testdata/ansible/memcached-operator/Makefile.bug2037208	2022-01-07 14:39:29.949793617 -0500
+++ ./testdata/ansible/memcached-operator/Makefile	2022-01-07 14:42:29.905225836 -0500
@@ -83,7 +83,7 @@ undeploy: ## Undeploy controller from th
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
 OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
-ARCH := $(shell uname -m | sed 's/x86_64/amd64/')
+ARCH := $(shell uname -m | sed 's/x86_64/amd64/' | sed 's/aarch64/arm64/')
 
 .PHONY: kustomize
 KUSTOMIZE = $(shell pwd)/bin/kustomize
diff -up ./testdata/helm/memcached-operator/Makefile.bug2037208 ./testdata/helm/memcached-operator/Makefile
--- ./testdata/helm/memcached-operator/Makefile.bug2037208	2022-01-07 14:39:52.522098536 -0500
+++ ./testdata/helm/memcached-operator/Makefile	2022-01-07 14:42:59.089620689 -0500
@@ -83,7 +83,7 @@ undeploy: ## Undeploy controller from th
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
 OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
-ARCH := $(shell uname -m | sed 's/x86_64/amd64/')
+ARCH := $(shell uname -m | sed 's/x86_64/amd64/' | sed 's/aarch64/arm64/')
 
 .PHONY: kustomize
 KUSTOMIZE = $(shell pwd)/bin/kustomize
