diff -up ./Makefile.bug2041679 ./Makefile
--- ./Makefile.bug2041679	2022-01-25 14:28:35.381883568 -0500
+++ ./Makefile	2022-01-25 14:28:40.001915459 -0500
@@ -9,7 +9,7 @@ export IMAGE_VERSION = v1.16.0
 export SIMPLE_VERSION = v1.16.0-ocp
 export GIT_VERSION = $(SIMPLE_VERSION)
 export GIT_COMMIT = $(shell git rev-parse HEAD)
-export K8S_VERSION = 1.21
+export K8S_VERSION = 1.22
 # TODO: bump this to 1.21, after kubectl `--generator` flag is removed from e2e tests.
 export ENVTEST_K8S_VERSION = 1.21.1
 
