diff -up ./go.mod.bug2099603 ./go.mod
--- ./go.mod.bug2099603	2022-06-21 15:25:13.532379361 -0400
+++ ./go.mod	2022-06-21 15:26:47.576644916 -0400
@@ -15,7 +15,7 @@ require (
 	github.com/onsi/gomega v1.18.1
 	github.com/operator-framework/api v0.15.0
 	github.com/operator-framework/helm-operator-plugins v0.0.12-0.20220613184440-7329cace347f
-	github.com/operator-framework/java-operator-plugins v0.5.1
+	github.com/operator-framework/java-operator-plugins v0.5.1-0.20220621191241-d474ba0c3de5
 	github.com/operator-framework/operator-lib v0.11.0
 	github.com/operator-framework/operator-manifest-tools v0.2.1
 	github.com/operator-framework/operator-registry v1.23.0
diff -up ./go.sum.bug2099603 ./go.sum
--- ./go.sum.bug2099603	2022-06-21 15:25:33.256435070 -0400
+++ ./go.sum	2022-06-21 15:30:14.195228446 -0400
@@ -869,8 +869,8 @@ github.com/operator-framework/api v0.15.
 github.com/operator-framework/api v0.15.0/go.mod h1:scnY9xqSeCsOdtJtNoHIXd7OtHZ14gj1hkDA4+DlgLY=
 github.com/operator-framework/helm-operator-plugins v0.0.12-0.20220613184440-7329cace347f h1:lS/IvqlvEQGIwXE0VlW+mOCmFEXBKywNbGQDrK++r/g=
 github.com/operator-framework/helm-operator-plugins v0.0.12-0.20220613184440-7329cace347f/go.mod h1:D7zPPwmIFBqHtWigU2iJiLuZ0v7hOJOb1/VC+/UuBAQ=
-github.com/operator-framework/java-operator-plugins v0.5.1 h1:HmiTocc61d/uqVPY/7EUR6ZTHDVeZ5/fgy7uo1QIBFc=
-github.com/operator-framework/java-operator-plugins v0.5.1/go.mod h1:UnUHAWY203Xw1j6Xpiirp/psJJaSRYcjenc0NH2+aVw=
+github.com/operator-framework/java-operator-plugins v0.5.1-0.20220621191241-d474ba0c3de5 h1:6Nam18EspFwdx/CoIO2vHvDncOKCvz7vOPy0uvZErDk=
+github.com/operator-framework/java-operator-plugins v0.5.1-0.20220621191241-d474ba0c3de5/go.mod h1:UnUHAWY203Xw1j6Xpiirp/psJJaSRYcjenc0NH2+aVw=
 github.com/operator-framework/operator-lib v0.11.0 h1:eYzqpiOfq9WBI4Trddisiq/X9BwCisZd3rIzmHRC9Z8=
 github.com/operator-framework/operator-lib v0.11.0/go.mod h1:RpyKhFAoG6DmKTDIwMuO6pI3LRc8IE9rxEYWy476o6g=
 github.com/operator-framework/operator-manifest-tools v0.2.1 h1:hD3iyOm2mBItzYhpFFWqU1StkolS4XGvXxRvFO4v3Oo=
diff -up ./vendor/github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/api.go.bug2099603 ./vendor/github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/api.go
--- ./vendor/github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/api.go.bug2099603	2022-06-21 15:35:41.571152992 -0400
+++ ./vendor/github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/api.go	2022-06-21 15:36:54.320358453 -0400
@@ -109,7 +109,7 @@ func (p *createAPISubcommand) Scaffold(f
 			projectName = strings.ToLower(filepath.Base(dir))
 		}
 
-		makefileBytes = append(makefileBytes, []byte(fmt.Sprintf(makefileBundleVarFragment, p.resource.Plural, p.resource.QualifiedGroup(), p.resource.Version, projectName))...)
+		makefileBytes = append(makefileBytes, []byte(fmt.Sprintf(makefileBundleVarFragment, p.resource.Plural, p.resource.QualifiedGroup(), p.resource.Version))...)
 
 		makefileBytes = append([]byte(fmt.Sprintf(makefileBundleImageFragement, p.config.GetDomain(), projectName)), makefileBytes...)
 
@@ -217,7 +217,7 @@ const (
 .PHONY: bundle
 bundle:  ## Generate bundle manifests and metadata, then validate generated files.
 ## marker
-	cat target/kubernetes/%[1]s.%[2]s-%[3]s.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=%[4]s
+	cat target/kubernetes/%[1]s.%[2]s-%[3]s.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	operator-sdk bundle validate ./bundle
 	
 .PHONY: bundle-build
@@ -233,7 +233,35 @@ bundle-push: ## Push the bundle image.
 const (
 	makefileBundleImageFragement = `
 VERSION ?= 0.0.1
+
+# CHANNELS define the bundle channels used in the bundle.
+# Add a new line here if you would like to change its default config. (E.g CHANNELS = "candidate,fast,stable")
+# To re-generate a bundle for other specific channels without changing the standard setup, you can:
+# - use the CHANNELS as arg of the bundle target (e.g make bundle CHANNELS=candidate,fast,stable)
+# - use environment variables to overwrite this value (e.g export CHANNELS="candidate,fast,stable")
+ifneq ($(origin CHANNELS), undefined)
+BUNDLE_CHANNELS := --channels=$(CHANNELS)
+endif
+
+# DEFAULT_CHANNEL defines the default channel used in the bundle.
+# Add a new line here if you would like to change its default config. (E.g DEFAULT_CHANNEL = "stable")
+# To re-generate a bundle for any other default channel without changing the default setup, you can:
+# - use the DEFAULT_CHANNEL as arg of the bundle target (e.g make bundle DEFAULT_CHANNEL=stable)
+# - use environment variables to overwrite this value (e.g export DEFAULT_CHANNEL="stable")
+ifneq ($(origin DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+# IMAGE_TAG_BASE defines the docker.io namespace and part of the image name for remote images.
+# This variable is used to construct full image tags for bundle and catalog images.
+#
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# %[1]s/%[2]s-bundle:$VERSION and %[1]s/%[2]s-catalog:$VERSION.
 IMAGE_TAG_BASE ?= %[1]s/%[2]s
+
+# BUNDLE_IMG defines the image:tag used for the bundle.
+# You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
 BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
 `
 )
diff -up ./vendor/modules.txt.bug2099603 ./vendor/modules.txt
--- ./vendor/modules.txt.bug2099603	2022-06-21 15:34:28.872947675 -0400
+++ ./vendor/modules.txt	2022-06-21 15:36:54.827359885 -0400
@@ -747,7 +747,7 @@ github.com/operator-framework/helm-opera
 github.com/operator-framework/helm-operator-plugins/pkg/plugins/hybrid/v1alpha/scaffolds/internal/templates/hack
 github.com/operator-framework/helm-operator-plugins/pkg/plugins/hybrid/v1alpha/scaffolds/internal/templates/rbac
 github.com/operator-framework/helm-operator-plugins/pkg/plugins/util
-# github.com/operator-framework/java-operator-plugins v0.5.1
+# github.com/operator-framework/java-operator-plugins v0.5.1-0.20220621191241-d474ba0c3de5
 ## explicit; go 1.17
 github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha
 github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/scaffolds
